smart-contracts/wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -496,29 +479,22 @@ impl<V> CachedRef<V> {
                 let reference = backing_store.store_raw(value.as_ref())?;
                 // Take the value out of the cachedref and temporarily replace it with
                 // a dummy value.
-                let value = std::mem::replace(self, CachedRef::Disk {
+                // and now write the cached value back in.
+                *self = CachedRef::Disk {
                     reference,
-                });
-                if let CachedRef::Memory {
-                    value,
-                } = value
-                {
-                    // and now write the cached value back in.
-                    *self = CachedRef::Cached {
-                        value,
-                        reference,
-                    };
-                } else {
-                    // this is unreachable since we hold a mutable reference to the cached value
-                    // and we know the value was a purely in-memory one
-                    unsafe { std::hint::unreachable_unchecked() }
-                }
+                };
                 reference.store(buf)
             }
             CachedRef::Cached {
-                reference,
-                ..
-            } => reference.store(buf),
+                reference: _,
+                value,
+            } => {
+                let reference = backing_store.store_raw(value.as_ref())?;
```

Comment:
I don't think you want to be storing the value again, right? Why not reuse the reference you already have?

Reply:
Excellent observation. Fixed now.